content/browser/loader/navigation_url_loader_impl.cc

```diff
@@ -977,7 +977,7 @@ void NavigationURLLoaderImpl::OnReceiveRedirect(
       (int) global_request_id_.request_id
     );
     dict.Set("requestId", request_id);
-    dict.Set("originalUrl", url_chain_[0].spec());
+    dict.Set("originalUrl", url_chain_.size() > 0 ? url_chain_[0].spec() : base::EmptyString());
```

Comment:
I think instead of `base::EmptyString()` we want `url_.spec()` here?

Reply:
We already have that in `dict.Set("requestUrl", url_.spec())`.

Reply:
It kinda feels liks there are bigger issues here, like for example, why do we take the first, rather than the last URL in `url_chain_`?

Reply:
Looking at the logic in the same function, `url_` seems to be the actual "original URL":

```cpp
GURL previous_url = url_;
url_ = redirect_info.new_url;
```

Reply:
> It kinda feels liks there are bigger issues here, like for example, why do we take the first, rather than the last URL in `url_chain_`?

I'm guessing the assumption here is that `originalUrl` is the one the user originally requested, not just the one we're currently requesting (and have received a redirect for).

> Looking at the logic in the same function, `url_` seems to be the actual "original URL":
> 
> ```c++
> GURL previous_url = url_;
> url_ = redirect_info.new_url;
> ```

Only for the most recent url request.  There can be more than one redirect (hence `url_chain_`).

I guess the question here is: who is the consumer of this data?  and what assumptions do they have about it?